docs/modules/ROOT/pages/crosschain.adoc

```diff
@@ -0,0 +1,72 @@
+= Cross-chain messaging
+
+Developers building contracts may require cross-chain functionality. To accomplish it, multiple protocols have implemented their own ways to process operations across chains.
+
+The variety of these bridges is outlined in https://x.com/norswap[@norswap]'s https://github.com/0xFableOrg/xchain/blob/master/README.md[Cross-Chain Interoperability Report] that proposes https://github.com/0xFableOrg/xchain/blob/master/README.md#bridge-taxonomy[a taxonomy of 7 bridge categories]. This diversity makes it difficult for developers to design cross-chain applications given the lack of portability.
+
+This guide will teach you how to follow ERC-7786 to establish messaging gateways across chains regardless of the underlying bridge. Developers can implement gateway contracts that process cross-chain messages and connect any crosschain protocol they want (or implement themselves).
```

Comment:
This part makes me wonder who is this guide for ? Devs that will build cross-chain apps or devs that will implement the gateway / gateway adatpers ?

Reply:
I'd say for devs building cross-chain apps. However, I think it's normal for both to think about connecting an underlying crosschain protocol. I wanted the guide to show how cross-chain communication is built from the ground up and then provide an out-of-the-box adapter like Axelar's

Reply:
IMO the whole point of 7786 is that the devs of apps taht perform crosschain operation don't have to implement anything that is specific to underlying protocol. in 95% of the case, they should not be implementing the gateway contracts themselves.

Reply:
Would you mind making a suggestion? I don't see how the current code is avoiding the use case you mention. If it's about how it's written, a suggestion would make sense. If it's related to the content, I don't understand if the current formulation is an issue.